src/index-worker.ts

```diff
@@ -88,20 +89,31 @@ ipcRenderer.on("edit-repo", (e: IpcRendererEvent, args: { id: string, repoName:
     repStore.update(id, repoName);
     ipcRenderer.sendTo(mainWindowId, "refresh-repos", getRepos());
 });
-ipcRenderer.on("test-perforce-connection", (e: IpcRendererEvent, connectionString: string) => {
+ipcRenderer.on("test-perforce-connection", (e: IpcRendererEvent, connectionOptions: PerforceConnectionOptions) => {
     let isSuccess = false;
     try {
-        isSuccess = !!changePerforceManagerSingleton(connectionString);
+        isSuccess = !!changePerforceManagerSingleton(connectionOptions);
     } catch (e) {
         if (e.message?.code === "ENOENT") {
             ipcRenderer.send("no-p4-found");
         }
-        console.error(`Failed to init perforce manager with port :${connectionString}`);
+        console.error(`Failed to init perforce manager with port :${connectionOptions}`);
     }
 
     ipcRenderer.sendTo(mainWindowId, "test-perforce-connection-result", isSuccess);
 });
 
+ipcRenderer.on("test-git-connection", async (e: IpcRendererEvent, connectionOptions: GitConnectionOptions) => {
+    let isSuccess = false;
+    try {
+        isSuccess = !!(await cloneRemoteOriginWithCommit(connectionOptions.connectionString, "master", false));
+    } catch (e) {
+        console.error(`Failed to clone git repo ${connectionOptions.connectionString}`, e);
```

Comment:
notify?

Reply:
Not sure this is not part of a normal flow though. This might be because of something we know of like credentials